eip-0042.md

```diff
@@ -0,0 +1,174 @@
+- A byte with the value `98` indicates **"atLeast"**.
+- Two bytes, with values `73` and `00`, respectively.
+  The value `00` indicates the VLQ-encoded index of value `N` in the list of constants.
+- A constant byte with value `83`, the VLQ-encoded value of `M`, and a constant byte with value `08`
+- For each public key (`M` times), a byte with value `73`,followed by the VLQ-encoded value of the index of the public key in the list of constants
+
+Therefore, for each derivation path, all wallets are able to generate a unique and similar byte array as an ErgoTree.
+
+![Ergo Tree Bytes](./eip-0042/bytes.png)
+
+## Transaction Signing
+
+An unsigned transaction is generated on one wallet. This transaction may have been generated through ErgoPay or the dApp connector, or it could be an outgoing transaction to another address through the wallet itself.
+
+After an unsigned transaction is generated, the wallet
+
+Generates a commitment.
+Stores its private part locally.
+Shares a commitment as below
+
+```
+{
+  tx:<reduced transaction bytes encoded as base64>,
+  boxes:[encoded boxes as base64],
+  commitment: commitments encoded
+}
+```
+
+The input `boxes:` are only used to display transactions on wallets.
+
+The general commitment format is as follows. However, this general-purpose commitment can be used for any contract.
+```
+{
+    "hint": "cmtReal",
+    "position": "0-0",
+    "type": "dlog",
+    "pubkey": {
+      "op": "205",
+      "h": "03a73c66970f14fc6450c6ab1a167cb4ba3baa64b20f731e22ec6840c70d27ef1c",
+    }
+    "a": "0253db866791af521ba4ab009509b6db89d272d9461636ee65eaa3e316884b21a4"
+}
+```
+
+In a multi-signature wallet, a simplified version of commitment is used.
+
+```
+commitments: [
+  [<base64 string of commitment for public key index 0 in input 0>, ...]
+  [<base64 string of commitment for public key index 0 in input 1>, ...]
+]
+```
+
+In the multi-signature wallet, a commitment is represented as a list of base64 strings to reduce the commitment size. Also, an empty string is used wherever a commitment is unavailable.
+
+[//]: # (and we can transform these commitments to standard format:)
+This reduced format can be transformed into the standard format:
+
+```
+{
+    "hint": "cmtReal",
+    "position": "0-<commitment_index>",
+    "type": "dlog",
+    "pubkey": {
+        "op": "205",
+        "h": "<public key used in address generation at index commitment_index>"
+    },
+    "a": "<base16 encoded of commitment at index commitment_index>"
+}
+```
+
+[//]: # (Because wallets want to pass this code via QRCode, each wallet can split it into chunks)
+
+To transfer these commitments using QRCode, we can split the data into several chunks:
+
+```
+{
+   MSR: "{\"tx\": ...}",
+   n: <number of chunks>
+   p: <page number>
+}
+```
+
+This data is transferred between multi-signature wallets until one wallet has at least `N` commitments (including its own).
+
+This wallet creates a partially signed transaction with this encoding:
```

Comment:
Why does the last wallet create a "partially signed transaction" and not a fully signed transaction if it already has N commitments? Or does this sentence apply until N-1 commitments are collected?

Reply:
see https://github.com/ergoplatform/eips/blob/9906f403f61d8592ab955494cc4310563f4067c6/eip-0011.md#signing-procedure. 
It's a two-round protocol in which the parties generate commitments in the first round. The last party is able to partially sign (he has only one of the keys so he cannot fully sign) and then, each of them signs the transaction partially and sends the generated signature to another party. This process continues until the signature is completed.